hotham/src/systems/audio.rs

```diff
@@ -18,32 +18,62 @@ pub fn audio_system(
     physics_context: &PhysicsContext,
     xr_context: &XrContext,
 ) {
-    for (_, (sound_emitter, rigid_body)) in query.query_mut(world) {
-        // First, where is the listener?
-        let listener_location =
-            get_location_from_poses(xr_context.views[0].pose, xr_context.views[1].pose);
+    // First, where is the listener?
+    let (listener_position_in_stage, listener_velocity_in_stage) = xr_context
+        .view_space
+        .relate(
+            &xr_context.stage_space,
+            xr_context.frame_state.predicted_display_time, // TODO: Use "now" instead.
+        )
+        .unwrap();
+
+    if !listener_position_in_stage
+        .location_flags
+        .contains(SpaceLocationFlags::POSITION_VALID)
+    {
+        return;
+    }
+
+    if !listener_velocity_in_stage
+        .velocity_flags
+        .contains(SpaceVelocityFlags::LINEAR_VALID)
+    {
+        return;
+    }
+
+    let listener_position_in_stage: Vector3<_> =
+        mint::Vector3::from(listener_position_in_stage.pose.position).into();
+    let listener_velocity_in_stage: Vector3<_> =
+        mint::Vector3::from(listener_velocity_in_stage.linear_velocity).into();
 
+    for (_, (sound_emitter, rigid_body)) in query.query_mut(world) {
         // Get the position and velocity of the entity.
         let rigid_body = physics_context
             .rigid_bodies
             .get(rigid_body.handle)
             .expect("Unable to get RigidBody");
 
-        let velocity = (*rigid_body.linvel()).into();
+        let source_position_in_stage: Point3<_> = (*rigid_body.translation()).into();
+        let source_velocity_in_stage: Vector3<_> = *rigid_body.linvel();
 
-        // Now transform the position of the entity w.r.t. the listener
-        let position = listener_location
-            .inverse_transform_point(&Point3::from(*rigid_body.translation()))
-            .into();
+        // Compute relative position and velocity
+        let relative_position_in_stage =
+            (source_position_in_stage - listener_position_in_stage).into();
+        let relative_velocity_in_stage =
+            (source_velocity_in_stage - listener_velocity_in_stage).into();
```

Comment:
There are lots of conversions going on here. Please let me know if you know how to make this more tidy!

Reply:
For me this is a case where explicit and verbose beats terse and complex! Happy to leave as is